### PR TITLE
Imporve git meta merge-bare #744

### PR DESCRIPTION
**Pull request by shijinglu**

### CODE REVIEW

File: node/lib/cmd/merge_bare.js [hunk 1]
```diff
@@ -62,19 +62,37 @@ exports.configureParser = function (parser) {
...
         help: "their side commitish to merge",
-        nargs: 1,
     });
 
     parser.addArgument(["--no-ff"], {
         help: "create a merge commit even if fast-forwarding is possible",
         action: "storeConst",
         constant: true,
     });
+    parser.addArgument(["--author-name"], {
```
Comment (novalis) [inline node/lib/cmd/merge_bare.js]: Regular git uses GIT_AUTHOR_NAME etc

File: node/lib/util/merge_common.js [hunk 2]
```diff
@@ -233,6 +241,32 @@ MergeContext.prototype.getSig = co.wrap(function *() {
     return yield ConfigUtil.defaultSignature(this.d_metaRepo);
 });
 
+/**
+ * @async
+ * @returns {NodeGit.Signature} author to be set with merge commit
+ */
+MergeContext.prototype.getAuthor = co.wrap(function *() {
+    if(this.d_authorName && this.d_authorEmail) {
```
Comment (novalis) [inline node/lib/util/merge_common.js]: nit: space before (

File: node/lib/util/merge_util.js [hunk 3]
```diff
@@ -98,17 +98,23 @@ const checkForMerge = co.wrap(function *(path) {
...
+const getBareMergeConflictsMessage = function(conflicts) {
     if (0 === Object.keys(conflicts).length) {
         return "";
     }
     let errorMessage = "CONFLICT (content): \n";
     const names = Object.keys(conflicts).sort();
     for (let name of names) {
-        errorMessage += `Conflicting entries for submodule: ` + 
-            `${colors.red(name)}\n`;
+        const conflict = conflicts[name];
```
Comment (novalis) [inline node/lib/util/merge_util.js]: Can you break this into two commits -- one for the conflicts change and the other for the author stuff?

File: node/test/util/merge_bare.js [hunk 4]
```diff
@@ -78,6 +78,26 @@ x=U:C3-2 s=Sa:a;C5-4 t=Sa:b;C4-2 s=Sa:b;Bmaster=3;Bfoo=5`,
...
+            "one merge with author and committer": {
+                initial: `
+a=B:Ca-1;Cb-1;Ba=a;Bb=b|
+x=U:C3-2 s=Sa:a;C4-2 s=Sa:b;Bmaster=3;Bfoo=4`,
+                theirCommit: "4",
+                ourCommit: "3",
+                authorName: "alice",
+                authorEmail: "alice@wonderland.com",
+                committerName: "bob",
+                committerEmail: "bob@twosigma.com",
```
Comment (novalis) [inline node/test/util/merge_bare.js]: Please use example.com addresses for all fake emails.

**Comment by shijinglu** [comment]
Now git meta merge-bare with conflicts looks like:

```
$ git meta merge-bare f1 f3 -m 'merge f1 f3'  
Submodule s1234: merging commit cbb2bde919d74ac44ac38f85b02db49c91e4dbf3.
Submodule s123: merging commit 5944ff17f5486f9418ee93f47b44c666e62bf16c.
CONFLICT (content):
	conflicted:  s123/README.md
	conflicted:  s1234/README.md

Automatic merge failed
```

without conflicts:
```
$ git meta merge-bare f0 f2 -m 'merge f0 f2' 2>/dev/null
c329628e583b2f0a9712bdd018d6272d267cb2f7
```

**Pull request merged**